### ISSUE TITLE
Show more source details in lsxfel

### ISSUE THREAD

**Issue by takluyver** [European-XFEL/EXtra-data#29]
`lsxfel` can list the sources in a file or a run. It should be possible to see more info about them - e.g. how much data is recorded for each,  what keys it contains, and the data shape of those keys.

Gathering this information may be slow, so we probably shouldn't do it by default. Two possibilities:

- A global flag like `-v` for verbose, or `--keys`, to show more information for all sources.
- An option to inspect specified sources more closely, e.g. `--source '*/XGM/*'`. Allowing wildcards is convenient to avoid typing/copying long source names, but you'd need single quotes to stop the shell trying to expand them.

**Comment by fangohr** [comment]
Yes, please! I think this is a really useful feature: it may be used at the end of a run to quickly check that the right sources are in the files, and also that data has been recorded for these sources. I think this is a step that is often done manually at the moment (using h5ls or h5view or so).

I think both the exhaustive `-v` option but also the ability to select a particular source or set of sources would be useful.

**Comment by takluyver** [comment]
I'm leaning towards the source option, because you could specify `--source '*'` to give more detail on all sources.

**Comment by fangohr** [comment]
It might still be attractive to have the option to write `-v` instead of `--source '*'` for convenience? Or we can wait for people to request it. Or use another flag. 

(But `--source '*'` does not seem natural to me to type if you need it more than once.)

**Comment by takluyver** [comment]
PR #38 would have the option called `--detail` instead, taking a source pattern, e.g. `--detail '*/XGM/*'`.

I'm open to adding a `--detail-all` flag or something, but I prefer to get a single option designed and working first. And showing the details for all sources is fairly slow and produces a lot of output, so I think it makes sense to start with the one that allows selecting sources.

**Comment by takluyver** [comment]
The `--detail` option described in the previous comment got merged for 1.2.

I'll leave this issue open for now to consider something like a `--detail-all` flag. But as in the previous comment, I'd like to get some experience using the `--detail` option before we make a decision on adding another option.
